sealedenumprocessor/src/main/kotlin/com/livefront/sealedenum/internal/spec/EnumForSealedEnumTypeSpec.kt

```diff
@@ -10,21 +11,23 @@ import javax.lang.model.element.TypeElement
 /**
  * A builder for an enum class for the given [sealedClass].
  *
- * Given the [ClassName] of the [sealedClass], the [TypeElement] for the [sealedClassCompanionObjectElement], and the
- * list of [ClassName]s for the sealed subclasses, [build] will generate a [TypeSpec] for an enum class
+ * Given the [ClassName] of the [sealedClass], the [TypeElement] for the [sealedClassCompanionObjectElement], and
+ * the list of [ClassName]s for the sealed subclasses, [build] will generate a [TypeSpec] for an enum class.
  *
  * The name of the created enum class will be the name of the sealed class concatenated with [enumPrefix] concatenated
  * with "Enum".
  */
 internal data class EnumForSealedEnumTypeSpec(
     private val sealedClass: SealedClass,
+    private val sealedClassVisibility: Visibility,
     private val sealedClassCompanionObjectElement: TypeElement,
     private val sealedObjects: List<SealedObject>,
     private val parameterizedSealedClass: TypeName,
     private val enumPrefix: String,
     private val sealedClassInterfaces: List<TypeName>
 ) {
     private val typeSpecBuilder = TypeSpec.enumBuilder(sealedClass.createEnumForSealedEnumName(enumPrefix))
+        .addModifiers(sealedClassVisibility.kModifier)
```

Comment:
So previously this was just always an implied `public` right?

Reply:
Correct, adding no modifiers resulted in an implied `public`. kotlinpoet is also smart enough to not emit the `public` in the generated code even when we do specify it.